### ISSUE TITLE
Remove the `ADIOS2_USE_MPI` in `io_adios2.cxx`

### ISSUE THREAD

**Issue by eschnett** [eschnett/CarpetX#47]
This is now handled by Roland's external library.

Maybe leave a comment in the source code, or an `#ifdef`?

**Comment by rhaas80** [comment]
I like the `#ifdef`. Either to define `ADIOS2_USE_MPI` (though I think redifining it with the same value is legal in C) or as an error check:

```c
#if (!defined(ADIOS2_USE_MPI) || !ADIOS2_USE_MPI)
#error "CarpetX requires and MPI enabled ADIOS2 library. Please compile ADIOS2 with MPI support and enable its use by defining ADIOS2_USE_MPI."
#endif
```

**Comment by eschnett** [comment]
Implemented in #59.
